linkerd/core/src/main/scala/io/buoyant/linkerd/ActiveTracer.scala

```diff
@@ -0,0 +1,170 @@
+package io.buoyant.linkerd
+
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.Dst
+import com.twitter.finagle.client.Transporter.EndpointAddr
+import com.twitter.finagle.naming.buoyant.DstBindingFactory
+import com.twitter.util.{Future, Stopwatch}
+import io.buoyant.linkerd.ActiveTracer.formatDelegation
+import io.buoyant.namer.{DelegateTree, Delegator}
+import io.buoyant.namer.DelegateTree._
+import io.buoyant.router.RoutingFactory.BaseDtab
+
+private[linkerd] case class RouterContext(
```

Comment:
any reason this needs to be a case class?  can this just be a method on RouterContextBuilder?

Reply:
I was futureproofing this functionality. I reason that there are cases where we may want a view of `RouterContext` and not necessarily represent that as a string. That way, you can pass around a `RouterContext` object to do whatever work you may need to do.